lib/runner.js

```diff
@@ -633,6 +634,16 @@ Runner.prototype.runSuite = function(suite, fn) {
 
   debug('run suite %s', suite.fullTitle());
 
+  if (!total && this.forbidOnly && hasOnly(suite)) {
```

Comment:
why `!total`?

Reply:
Because non-empty suite already checked in `runTest`.